Finished GGSmtp

teamagam3 opened the pull request.

teamagam3 committed:
Finished GGSmtp

teamagam3 committed:
Edited unit test

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GGSmtp/pull/1" target="_blank"><img src="https://app.review.ninja/59655385/pull/1/badge" alt="ReviewNinja"/>

teamagam3 committed:
Added missing files

BarAgam commented on GGSmtp/GGSmtp.Tests/DataResponderTests.cs:
```diff
@@ -0,0 +1,40 @@
+        [Test(Author = "Dor", Description = "Checks the DefaultDataResponder", TestOf = typeof(DefaultDataResponder)]
+        public void TestAttachmentSend()
+        {
+            string[] lines = File.ReadAllLines(@"C:\Temp\Mail\2016-05-15_113132_746.eml");
```
Accessing local files is problematic.
This test will fail on any new PC.
I'd hard-code the lines. !fix

teamagam3 replied:
It can be a lot of lines, we won't able to read the code.
What about test dataset?

BarAgam commented on GGSmtp/GGSmtp.Tests/DataResponderTests.cs:
```diff
@@ -0,0 +1,40 @@
+    [TestFixture]
+    public class DataResponderTests
+    {
+        [Test(Author = "Dor", Description = "Checks the DefaultDataResponder", TestOf = typeof(DefaultDataResponder)]
```
Missing parenthesis ')'
It doesn't compile !fix

teamagam3 replied:
!fixed

BarAgam commented on GGSmtp/GGSmtp.Tests/DataResponderTests.cs:
```diff
@@ -0,0 +1,40 @@
+namespace GGSmtp.Tests
+{
+    [TestFixture]
+    public class DataResponderTests
```
Our conventions might not be that clear, but a unit test should be named after the class under test with a Test(s) suffix.
So it should be named DefaultDataResponderTests !fix

teamagam3 replied:
!fixed

teamagam3 replied:
!fix

BarAgam commented on GGSmtp/GGSmtp.Tests/DataResponderTests.cs:
```diff
@@ -0,0 +1,40 @@
+    public class DataResponderTests
+    {
+        [Test(Author = "Dor", Description = "Checks the DefaultDataResponder", TestOf = typeof(DefaultDataResponder)]
+        public void TestAttachmentSend()
```
look at (or talk wit me about) our test-methods naming and arrangement conventions !fix

BarAgam commented on GGSmtp/GGSmtp/DefaultDataResponder.cs:
```diff
@@ -0,0 +1,153 @@
+            string successMessage = string.Format("{0} bytes received", streamLength);
+
+            // Write a success response to return, but don't return it yet
+            SmtpResponse response = SmtpResponses.OK.CloneAndChange(successMessage);
```
Can be moved to method's last return statement !fix

teamagam3 replied:
!fixed

BarAgam commented on GGSmtp/GGSmtp/DefaultDataResponder.cs:
```diff
@@ -0,0 +1,153 @@
+                {
+                    mLogger.Debug(string.Format("Got attachment from {0}, sending . . .", sessionInfo.MailFrom));
+
+                    bool success = SendAttachment(sessionInfo.MailFrom.MailAddress, attatchment);
```
Send attachment can be turned to void and throw an exception on failure.
It's wrapped in a try-catch anyways, and there's code duplication dealing it's failure !resolve

BarAgam replied:
You dont have to refactor it that way, It's just a suggestions.
You can avoid the code dups in other ways. perhaps a single return statement suffice

teamagam3 replied:
!resolved

BarAgam commented on GGSmtp/GGSmtp/DefaultDataResponder.cs:
```diff
@@ -0,0 +1,153 @@
+                // Read the mail message from the stream
+                MimeMessage message = MimeMessage.Load(currentStream);
+
+                foreach (MimePart attatchment in message.Attachments)
```
Should we expect more than one attachment?

teamagam3 replied:
I haven't tested the photos bulk, so maybe we should expect that.
Either way, it's not a bad thing to support it.

BarAgam commented on GGSmtp/GGSmtp/DefaultDataResponder.cs:
```diff
@@ -0,0 +1,153 @@
+                return SmtpResponses.InternalServerError;
+            }
+
+            currentStream.Dispose();
```
Dispose object at finally block to avoid memory leaks on caught exceptions !fix

teamagam3 replied:
!fixed

BarAgam commented on GGSmtp/GGSmtp/DefaultDataResponder.cs:
```diff
@@ -0,0 +1,153 @@
+
+            try
+            {
+                currentStream.Seek(0, SeekOrigin.Begin);
```
I'd refactor the inners of the try clause into a method ("forward attachment(s)". Might improve readability. Up to you.

teamagam3 replied:
!fixed

BarAgam commented:
Work with regions.

Static Members,Data members, Ctors, Public .. Private

BarAgam commented:
Remove unused imports.
Auto-Indent files.

teamagam3 committed:
CR Fixes

Added new method to forward attachments
Fixed possible of memory leak
Added regions
